runtime/doc/lua.txt

```diff
@@ -3296,6 +3296,23 @@ filter({f}, {src}, {...})                                  *vim.iter.filter()*
     See also: ~
       • |Iter:filter()|
 
+flatten({src}, {depth}, {...})                            *vim.iter.flatten()*
```

Comment:
This is not worth adding as a `vim.iter` function. We can always change that later, but for now we should not enter this one-way door. `vim.iter(...):flat():totable()` is perfectly reasonable.